test-resources/headless-core-stub/utils/src/jwks-cache-control/index.ts

```diff
@@ -0,0 +1,32 @@
+import { JSONWebKeySet } from "jose";
+
+let cachedJWKS: JSONWebKeySet | null = null;
+let cachedJWKSExpiry: number | null = null;
+
+export const isJWKSCacheValid = () => Boolean(cachedJWKS && cachedJWKSExpiry && cachedJWKSExpiry >= Date.now());
+
+export const parseCacheControlMaxAge = (cacheControlHeaderValue?: string): number => {
+    const match = cacheControlHeaderValue?.match(/max-age=(\d+)/);
+    const maxAgeSeconds = match ? parseInt(match[1], 10) : -1;
+    return Date.now() + maxAgeSeconds * 1000;
+};
+
+export const fetchAndCacheJWKS = async (jwksUrl: URL, logger: { info: (msg: string) => void }) => {
+    const jwksResponse = await fetch(jwksUrl);
+    if (!jwksResponse.ok) {
+        throw new Error(
+            `Error received from the JWKS endpoint, status received: ${jwksResponse.status} ${jwksResponse.statusText}`,
+        );
+    }
+
+    cachedJWKS = (await jwksResponse.json()) as JSONWebKeySet;
+    cachedJWKSExpiry = parseCacheControlMaxAge(jwksResponse?.headers?.get("Cache-Control") as string);
+    logger.info(`JWKS cache has been updated to  ${cachedJWKSExpiry}`);
+};
+
+export const clearJWKSCache = () => {
+    cachedJWKS = null;
+    cachedJWKSExpiry = null;
+};
+
+export const getCachedJWKS = () => cachedJWKS;
```

Comment:
Does this really need to be a method?`if(cachedJWKS)` is the same amount of code and means I don't have to navigate to another file to try and find if it does anything special

Reply:
I suppose the idea is that `jwks-cache-control` owns `cachedJWKS` which is not exported of like private hence the `getCachedJWKS`

Reply:
Resolving as discussed